### ISSUE TITLE
Poor generalization

### ISSUE THREAD

**Issue by buptlittlecabbage** [SynodicMonth/ChebyKAN#9]
I tried using ChebyKAN to train signal waveforms, but it showed poor generalization. What may be the reason？？
![image](https://github.com/SynodicMonth/ChebyKAN/assets/137387186/dc9a32ee-a567-44f4-aaf2-7c6f16cc9a5f)
is train data
![image](https://github.com/SynodicMonth/ChebyKAN/assets/137387186/4139ecf3-1070-4495-b489-0d8b82332dc2)
is test data.

**Comment by SynodicMonth** [comment]
Whats the structure of ur model? How is ChebyKAN used? Have u tried normal MLP/original KAN? 
Could you please provide more detailed information or code snippets to further investigate the problem?

**Comment by buptlittlecabbage** [comment]
Yes, I use regular MLP, but I haven't used the original KAN before. My model is ChebyKAN based DeepOKAN. In fact, it's a bit complex, so later I only used 4-layer ChebyKAN for simple input-output waveform mapping, and it didn't have good generalization. The relevant codes and results are as follows:
![image](https://github.com/SynodicMonth/ChebyKAN/assets/137387186/10ff85ef-dfee-4e16-9b8c-72c5e0b021d5)
![image](https://github.com/SynodicMonth/ChebyKAN/assets/137387186/a3d19741-d842-4dc6-8b66-f6a70ef36f03)
![4da5f5eadd63405dd35e75ff446ea46](https://github.com/SynodicMonth/ChebyKAN/assets/137387186/34e49a77-7190-4b37-aa39-f63ee58a9862)
is test data, train data is similar to the blue line. 
Later, I made some fine-tuning to the parameters and improved their generalization ability, but it seems that they are still not as good as MLP.
![00ad9ec04f9ef6f216e3ff5f874183f](https://github.com/SynodicMonth/ChebyKAN/assets/137387186/79d415a7-d24f-46e1-b1ad-fbd24fd78aa0)

**Comment by buptlittlecabbage** [comment]
The generalization of MLP allows two lines to basically overlap. So I want to ask, how can adjusting parameters enhance the generalization of ChebyKAN

**Comment by SynodicMonth** [comment]
Not sure about the reason. 
maybe the degree is too high, i found somewhere between 4-8 is enough, and higher degree makes it harder to train.
also i found ChebyKAN needs a really low lr if using Adam, (like 1e-4).
some found it better to use multiple different polynomials (https://github.com/SynodicMonth/ChebyKAN/issues/4#issuecomment-2111056963), not sure if it can help.

**Comment by buptlittlecabbage** [comment]
Thanks for your reply, and I hope that we can find more application of ChebyKAN.

**Comment by buptlittlecabbage** [comment]
Thanks for your reply, and I hope that we can find more application of ChebyKAN.
